internal/lightcone/erudition/aninstantbeforeagaze/aninstantbeforeagaze.go

```diff
@@ -0,0 +1,58 @@
+package aninstantbeforeagaze
+
+import (
+	"github.com/simimpact/srsim/pkg/engine"
+	"github.com/simimpact/srsim/pkg/engine/equip/lightcone"
+	"github.com/simimpact/srsim/pkg/engine/event"
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+const (
+	AnInstantBeforeAGaze = "an-instant-before-a-gaze"
+)
+
+// Increases the wearer's CRIT DMG by 36%.
+// When the wearer uses Ultimate, increases the wearer's Ultimate DMG based on their Max Energy.
+// Each point of Energy increases the Ultimate DMG by 0.36%, up to 180 points of Energy.
+func init() {
+	lightcone.Register(key.AnInstantBeforeAGaze, lightcone.Config{
+		CreatePassive: Create,
+		Rarity:        5,
+		Path:          model.Path_ERUDITION,
+		Promotions:    promotions,
+	})
+
+	modifier.Register(AnInstantBeforeAGaze, modifier.Config{
+		Listeners: modifier.Listeners{
+			OnBeforeHit: onBeforeHit,
+		},
+	})
+}
+
+func Create(engine engine.Engine, owner key.TargetID, lc info.LightCone) {
+	amt := 0.30 + 0.06*float64(lc.Imposition)
+	engine.AddModifier(owner, info.Modifier{
+		Name:   AnInstantBeforeAGaze,
+		Source: owner,
+		Stats:  info.PropMap{prop.CritDMG: amt},
+		State:  float64(lc.Imposition),
+	})
+}
+
+func onBeforeHit(mod *modifier.Instance, e event.HitStart) {
```

Comment:
in game implementation has this done as two modifiers:
1. adds modifier in `OnBeforeSkillUse`/`OnActionStart` if action is ult
2. This modifer listens to `OnBeforeHit` and buffs the damage of ult hits
3. Modifier is removed `OnAfterSkillUse`/`OnActionEnd`

In the current state of sim, your implementation will have the desired behavior; however, in the future this may change once we support and understand multi attack types better

Am fine with this for now (this is the more efficient approach). Leave it up to you if you want to change it to be technically more "game accurate."